packages/blockchain-wallet-v4-frontend/src/data/components/identityVerification/sagas.js

```diff
@@ -0,0 +1,93 @@
+import { put, select } from 'redux-saga/effects'
+import { actions, selectors } from 'data'
+
+import * as A from './actions'
+import {
+  PERSONAL_STEPS,
+  EMAIL_STEPS,
+  SMS_STEPS,
+  EMAIL_FORM,
+  SMS_NUMBER_FORM
+} from './model'
+
+export const logLocation = 'components/identityVerification/sagas'
+
+export default ({ api, coreSagas }) => {
+  const derivePersonalStep = ({
+    email,
+    emailVerified,
+    smsNumber,
+    smsVerified
+  }) => {
+    if (!email || !emailVerified) return PERSONAL_STEPS.email
+    if (!smsNumber || !smsVerified) return PERSONAL_STEPS.smsNumber
+    return PERSONAL_STEPS.personal
+  }
+
+  const deriveEmailStep = ({ email, emailVerified }) => {
+    if (email && !emailVerified) return EMAIL_STEPS.verify
+    return EMAIL_STEPS.edit
+  }
+
+  const deriveSmsStep = ({ smsNumber, smsVerified }) => {
+    if (smsNumber && !smsVerified) return SMS_STEPS.verify
+    return SMS_STEPS.edit
+  }
+
+  const updatePersonalStep = function*(action) {
+    const { email, emailVerified, smsNumber, smsVerified } = action.payload
+    yield put(
+      A.setPersonalStep(
+        derivePersonalStep({ email, emailVerified, smsNumber, smsVerified })
+      )
+    )
+    yield put(A.setEmailStep(deriveEmailStep({ email, emailVerified })))
+    yield put(A.setSmsStep(deriveSmsStep({ smsNumber, smsVerified })))
+  }
+
+  const updateEmail = function*(args) {
+    const { email } = yield select(selectors.form.getFormValues(EMAIL_FORM))
+    yield put(actions.modules.securityCenter.updateEmail(email))
```

Comment:
Will this trigger confirmation code email resend? It should

Reply:
yes it will